modules/opentracing/src/main/scala/OTTracer.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez
+package opentracing
+
+import cats.effect.{Resource, Sync}
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import io.{opentracing => ot}
+import io.opentracing.propagation.{Format, TextMapAdapter}
+
+import scala.jdk.CollectionConverters._
+
+object OTTracer {
+
+  private[opentracing] def finishSpan[F[_] : Sync](t: ot.Tracer): ot.Span => F[Unit] = s =>
+    Sync[F].delay {
+      t.scopeManager().activate(s)
```

Comment:
I had to add this to get it working with XRay (which uses a ThreadLocal to track the active span in its SDK—afaict that can mostly be ignored, but this sets the span in the ThreadLocal before it's finished on the next line)

Full disclosure: I haven't tested it with Jaeger, but from some of the GitHub threads I was reading, it sounds like OpenTracing implementations are meant to use ScopeManager for this purpose anyway, so I'm hopeful it will be fine.